basic_autonomy/src/helper_functions.cpp

```diff
@@ -112,5 +112,28 @@ namespace waypoint_generation
         double ending_downtrack = wm->routeTrackPos(state_pos).downtrack;
         return get_nearest_index_by_downtrack(points, wm, ending_downtrack);
     }
+
+    cav_msgs::VehicleState get_nearest_state_to_downtrack(const std::vector<lanelet::BasicPoint2d>& points, double target_downtrack, const carma_wm::WorldModelConstPtr &wm)
```

Comment:
Doesn't a method in the world model do basically this, like get nearest index by downtrack or something like that?

Reply:
And this doesn't seem to be fully populating the VehicleState message either, so that may not be the correct data structure to use for just the raw X, Y data. I'd be inclined to use the WM function if it fits the right API.